### PR TITLE
Docstring pandas.data frame.diff

### PR DESCRIPTION
**Pull request by adatasetaday**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.diff" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
I think we just need a `.` at the end of this line https://github.com/pandas-dev/pandas/pull/20227/files#diff-1e79abbbdd150d4771b91ea60a4e1cc7R4739

LGTM otherwise.

**Comment by adatasetaday** [comment]
I can definitely do that, but it won't be consistent with the other functions' documentation.

I'm also adding a "See Also" to DataFrame.pct_change() which I didn't realize was pretty much the same function using % instead of absolute value.

**Comment by adatasetaday** [comment]
As dicsussed, the "." was added at the end of the versionadded and passed validation without any errors.

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4722,20 +4722,87 @@ def melt(self, id_vars=None, value_vars=None, var_name=None,
...
 
         Returns
         -------
         diffed : DataFrame
+
+        See Also
+        --------
+        DataFrame.pct_change: Percent change over given number of periods.
+        Series.diff: First discrete difference of object
+
```
Comment (jreback) [inline pandas/core/frame.py]: add DataFrame.shift

File: pandas/core/frame.py [hunk 2]
```diff
@@ -4722,20 +4722,87 @@ def melt(self, id_vars=None, value_vars=None, var_name=None,
...
+        ...                    'c': [1, 4, 9, 16, 25, 36]})
+        >>> df
+           a  b   c
+        0  1  1   1
+        1  2  1   4
+        2  3  2   9
+        3  4  3  16
+        4  5  5  25
+        5  6  8  36
+        >>> df.diff()
```
Comment (jreback) [inline pandas/core/frame.py]: add a blank line between cases

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by adatasetaday — COMMENTED** [review]
As requested added blank line between cases and added reference to DataFrame.shift in "See Also"

**Comment by TomAugspurger** [comment]
Thanks @adatasetaday!

**Pull request merged**